pkl-core/src/main/java/org/pkl/core/plugin/PklPlugin.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pkl.core.plugin;
+
+import java.util.Optional;
+
+/** Describes the public SPI for Pkl engine plugins */
+public interface PklPlugin {
```

Comment:
New lightweight plugin interface provided by `pkl-core` over an SPI; the hook points provided to plug-ins only include one so far, `onContextCreated`, which has an opportunity to modify the `org.graalvm...Context.Builder` before it is built into a `Context` instance and used by Pkl.